tests/components/bmw_connected_drive/test_config_flow.py

```diff
@@ -82,33 +83,26 @@ async def test_full_user_flow_implementation(hass):
         assert result2["title"] == FIXTURE_COMPLETE_ENTRY[CONF_USERNAME]
         assert result2["data"] == FIXTURE_COMPLETE_ENTRY
 
-        assert len(mock_setup.mock_calls) == 1
         assert len(mock_setup_entry.mock_calls) == 1
 
 
 async def test_full_config_flow_implementation(hass):
-    """Test registering an integration and finishing flow works."""
+    """Test registering an integration from YAML and raising an error."""
     with patch(
         "bimmer_connected.account.ConnectedDriveAccount._get_vehicles",
         return_value=[],
     ), patch(
-        "homeassistant.components.bmw_connected_drive.async_setup", return_value=True
-    ) as mock_setup, patch(
         "homeassistant.components.bmw_connected_drive.async_setup_entry",
         return_value=True,
     ) as mock_setup_entry:
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN,
-            context={"source": config_entries.SOURCE_IMPORT},
-            data=FIXTURE_USER_INPUT,
-        )
-
-        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-        assert result["title"] == FIXTURE_IMPORT_ENTRY[CONF_USERNAME]
-        assert result["data"] == FIXTURE_IMPORT_ENTRY
+        with pytest.raises(UnknownStep):
+            await hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
```

Comment:
We can remove this test. It's not possible anymore.

Reply:
Just removed it.

My original intention was to test for the UnkownStep exception, as that shouldn't work anymore.
But you are right, if it never will be called it makes sense to remove that test.